### PR TITLE
Fix 07

### PR DESCRIPTION
**Pull request by matbesancon**
Seemed like CPLEX.jl needs a refresh for 0.7, I fixed some things.
Things I couldn't get right:
- The finalizer for `Env` needs to be redefined
- One `ccall` is done incorrectly, yielding an error in the tests
- `importall` is used, I haven't got through all places where importall-ed package are used

### CODE REVIEW

**Comment by mlubin** [comment]
Thanks! We don't have CI set up so this is hard to test (I don't have a CPLEX license). Could you confirm that CPLEX passes tests on both 0.6 and 0.7 under this PR? We are not dropping 0.6 support yet.

File: src/CPLEX.jl [hunk 1]
```diff
@@ -119,6 +120,8 @@ module CPLEX
         try
             eval(current_module(), Expr(:import,:JuMP))
             include("JuMPfunctions.jl")
+		catch e
```
Comment (mlubin) [inline src/CPLEX.jl]: Looks like a whitespace mismatch

File: test/runtests.jl [hunk 2]
```diff
@@ -1,4 +1,4 @@
-using Base.Test
+using Test
```
Comment (mlubin) [inline test/runtests.jl]: Should this be Compat.Test?

File: src/cpx_callbacks.jl [hunk 3]
```diff
@@ -19,8 +19,8 @@ function setcallbackcut(cbdata::CallbackData, where::Cint, ind::Vector{Cint}, va
     @assert length(val) == len
     sns = convert(Cint, sense)
     stat = @cpx_ccall(cutcallbackadd, Cint, (
-                      Ptr{Void},
-                      Ptr{Void},
+                      Ptr{Nothing},
```
Comment (mlubin) [inline src/cpx_callbacks.jl]: Please use `Cvoid` instead of `Nothing` for `void*` types.

File: test/runtests.jl (hunk 2 above)
Comment (matbesancon) [reply test/runtests.jl]: this looks fine with just "Test", it was suggested directly by the compiler

**Comment by matbesancon** [comment]
As mentioned, there is still a ccall yielding an error and I could not track down where it's coming from
```
ERROR: LoadError: LoadError: syntax: ccall argument types must be a tuple; try "(T,)" and check if you specified a correct return type
Stacktrace:
 [1] include at ./boot.jl:317 [inlined]
 [2] include_relative(::Module, ::String) at ./loading.jl:1034
 [3] include at ./sysimg.jl:29 [inlined]
 [4] include(::String) at /home/mathieu/.julia/dev/CPLEX/src/CPLEX.jl:3
 [5] top-level scope at none:0
 [6] include at ./boot.jl:317 [inlined]
 [7] include_relative(::Module, ::String) at ./loading.jl:1034
 [8] include(::Module, ::String) at ./sysimg.jl:29
 [9] top-level scope at none:0
 [10] eval at ./boot.jl:319 [inlined]
 [11] eval(::Expr) at ./client.jl:394
 [12] top-level scope at ./none:3 [inlined]
 [13] top-level scope at ./<missing>:0
in expression starting at /home/mathieu/.julia/dev/CPLEX/src/cpx_solve.jl:1
in expression starting at /home/mathieu/.julia/dev/CPLEX/src/CPLEX.jl:112
```

File: src/CPLEX.jl (hunk 1 above)
Comment (matbesancon) [reply src/CPLEX.jl]: I got tricked by a tab-space conversion

File: src/cpx_callbacks.jl (hunk 3 above)
Comment (matbesancon) [reply src/cpx_callbacks.jl]: fixed in last commit

File: src/CPLEX.jl (hunk 1 above)
Comment (matbesancon) [reply src/CPLEX.jl]: fixed in last commit

**Pull request merged**